bpf/lib/nodeport.h

```diff
@@ -1134,12 +1137,10 @@ static __always_inline int nodeport_lb6(struct __ctx_buff *ctx,
 		if (nodeport_uses_dsr6(&tuple)) {
 #if (defined(IS_BPF_OVERLAY) && DSR_ENCAP_MODE == DSR_ENCAP_GENEVE) || \
 	(!defined(IS_BPF_OVERLAY) && DSR_ENCAP_MODE != DSR_ENCAP_GENEVE)
-			bool dsr = false;
-
 			ret = nodeport_extract_dsr_v6(ctx, ip6, &tuple, l4_off,
 						      &key.address,
-						      &key.dport, &dsr);
-			if (dsr) {
+						      &key.dport, dsr);
+			if (*dsr) {
```

Comment:
Just spotted this while driving-by, and think the approach we took in `main` won't work for `v1.14`. Probably best to drop https://github.com/cilium/cilium/pull/29155 for now, sorry :/.

`main` has https://github.com/cilium/cilium/pull/27267 and returns inline. `v1.14` returns via tail-call, so the `dsr` flag will never propagate back.

@ysksuzuki @chez-shanpu fyi. Think we need to check if we can build a custom backport.

Reply:
Ah, right. Will look into that.

Reply:
Let's drop #29155 from this PR. I will raise a PR against v1.14 later. Is that OK with you? Or do you plan to send a PR? @chez-shanpu

Reply:
> Let's drop #29155 from this PR. I will raise a PR against v1.14 later. Is that OK with you? Or are you planning to send a PR? @chez-shanpu

@ysksuzuki No problem, thank you :)